From 5a67f8f9d3a7f80a49554fb1ca4e68a1effd5399 Mon Sep 17 00:00:00 2001
From: Stephen Hemminger <stephen.hemminger@vyatta.com>
Date: Mon, 15 Sep 2008 12:05:11 -0700
Subject: [PATCH] Update to 2.6.27 API

The one issue was the old multiqueue API, so that is handled
by tc_util.h
---
 include/linux/atm.h                      |    1 +
 include/linux/if.h                       |   15 ++++++++-------
 include/linux/if_vlan.h                  |    1 +
 include/linux/ip6_tunnel.h               |    4 ----
 include/linux/netfilter.h                |    1 +
 include/linux/netfilter_ipv4.h           |    1 +
 include/linux/netfilter_ipv4/ip_tables.h |    1 +
 include/linux/pkt_cls.h                  |    1 +
 include/linux/pkt_sched.h                |   29 ++++++++++++++++++++---------
 include/linux/rtnetlink.h                |    1 +
 include/linux/socket.h                   |    5 ++++-
 tc/tc_util.h                             |   12 ++++++++++++
 12 files changed, 51 insertions(+), 21 deletions(-)

diff --git a/tc/tc_util.h b/tc/tc_util.h
index 796da54..39d5367 100644
--- a/tc/tc_util.h
+++ b/tc/tc_util.h
@@ -7,6 +7,18 @@
 #include <linux/gen_stats.h>
 #include "tc_core.h"
 
+/* This is the deprecated multiqueue interface */
+#ifndef TCA_PRIO_MAX
+enum
+{
+	TCA_PRIO_UNSPEC,
+	TCA_PRIO_MQ,
+	__TCA_PRIO_MAX
+};
+
+#define TCA_PRIO_MAX    (__TCA_PRIO_MAX - 1)
+#endif
+
 struct qdisc_util
 {
 	struct  qdisc_util *next;
-- 
1.6.0.2

